freetextresponse/freetextresponse.py

```diff
@@ -0,0 +1,437 @@
+'''
+This is the core logic for the FreeText XBlock
+'''
+
+import os
+import re
+
+import pkg_resources
+
+from enum import Enum
+from xblock.core import XBlock
+from xblock.fields import Scope
+from xblock.fields import String
+from xblock.fields import Integer
+from xblock.fields import Float
+from xblock.fragment import Fragment
+
+
+class Freetextresponse(XBlock):
+    #  pylint: disable=too-many-ancestors, too-many-instance-attributes
+    '''
+    Enables instructors to create questions with free-text responses.
+    '''
+    @staticmethod
+    def workbench_scenarios():
+        '''
+        Gather scenarios to be displayed in the workbench
+        '''
+        return [
+            ('FreeText XBlock',
+             '''<sequence_demo>
+                    <freetextresponse />
+                    <freetextresponse name="My First XBlock" />
+                </sequence_demo>
+             '''),
+        ]
+    display_name = String(
+        default='FreeText XBlock',
```

Comment:
Just a thought, is there maybe a better default?

At the least, I think we should change "XBlock"; instructors don't care if it's an XBlock/XModule/CAPA-problem/LTI-component/etc. and students will have no idea. Maybe something like "Free-text Response" is suitably more general?

Ideally, instructors would give everything an apt display name, but realistically many won't. This way, our default case is less cryptic to non-developers.